python/ray/autoscaler/_private/node_launcher.py

```diff
@@ -69,9 +82,12 @@ def run(self):
             try:
                 self._launch_node(config, count, node_type)
             except Exception:
+                if self.prom_metrics:
+                    self.prom_metrics.node_launch_exceptions.inc()
                 logger.exception("Launch failed")
             finally:
                 self.pending.dec(node_type, count)
+                self.prom_metrics.pending_nodes.set(self.pending.value)
```

Comment:
```suggestion
				if self.prom_metrics:
                    self.prom_metrics.pending_nodes.set(self.pending.value)
```

Reply:
Woah, really good catch!

Reply:
Although maybe at this point the safer thing to do is to just construct a default value for self.prom_metrics if none is passed in

Reply:
resolved indirectly on line 31